tests/atoms/gaia_apps.js

```diff
@@ -181,20 +181,28 @@ var GaiaApps = {
     var apps = window.wrappedJSObject.applications ||
                 window.wrappedJSObject.Applications;
     var app = apps.getByManifestURL(manifestURL);
-    var appName, launchPath;
-
-    if (entryPoint) {
-      if (app.manifest.entry_points[entryPoint]) {
-        appName = app.manifest.entry_points[entryPoint].name;
-        launchPath = app.manifest.entry_points[entryPoint].launch_path;
-      } else {
-        app = null;
+    var appManifestURL, launchPath;
+
+      if(app != null){
+        if (entryPoint) {
```

Comment:
This function has a depth of 3. That is to say, there are 3 if/else, one in another. This can be problematic, because this function alone covers 8 (2^3) different scenarios. I'd suggest to break it down. For example:
```js
if (app != null) {
   GaiaApps._formatApp(app, entryPoint, callback)
} else {
   callback(false);
} 
```
and further below:
```js
_formatApp: function(app, manifestUrl entryPoint, aCallback) {
    if (entryPoint) {
        if (app.manifest.entry_points[entryPoint]) {
            launchPath = app.manifest.entry_points[entryPoint].launch_path;
        } else {
            app = null;
        }
    } else {
        launchPath = app.manifest.launch_path;
    }
    GaiaApps.sendLocateResponse(aCallback, app, manifestURL, launchPath, entryPoint); 
}
```